File: packages/py/src/br_kit/cep.py
"""CEP (postal code) structural validation and formatting. Mirrors ``cep.ts``."""

from __future__ import annotations

import re

_MASK = re.compile(r"[.\-\s]")
# [0-9] (not \d) to match JavaScript's ASCII \d exactly (parity with cep.ts).
_ONLY_DIGITS = re.compile(r"^[0-9]{8}$")


def strip(value: str) -> str:
    """Remove mask characters (dot, hyphen, whitespace), keeping only digits."""
    return _MASK.sub("", value)


def is_valid(value: str) -> bool:
    """Structural validation only: a CEP is eight digits.

    Existence of the postal code is out of scope (no network calls in the core).
    """
    return bool(_ONLY_DIGITS.match(strip(value)))


def format(value: str) -> str:  # noqa: A001 - mirrors the cross-language API name
    """Format eight digits as ``00000-000``. Raises on a non-8-digit input."""
    c = strip(value)
    if not _ONLY_DIGITS.match(c):
        raise ValueError(f"Invalid CEP: cannot format {value!r}")
    return f"{c[:5]}-{c[5:]}"

File: packages/ts/src/cep.ts
const MASK = /[.\-\s]/g;
const ONLY_DIGITS = /^\d{8}$/;

/** Remove mask characters (dot, hyphen, whitespace), keeping only digits. */
export function strip(value: string): string {
  return value.replace(MASK, "");
}

/**
 * Structural validation only: a CEP is eight digits. Existence of the postal
 * code is out of scope (no network calls in the core — see design principle 6).
 */
export function isValid(value: string): boolean {
  return ONLY_DIGITS.test(strip(value));
}

/** Format eight digits as `00000-000`. Throws on a non-8-digit input. */
export function format(value: string): string {
  const c = strip(value);
  if (!ONLY_DIGITS.test(c)) {
    throw new RangeError(`Invalid CEP: cannot format ${JSON.stringify(value)}`);
  }
  return `${c.slice(0, 5)}-${c.slice(5)}`;
}
